File: languages/HamsterSimulatorLanguage.sandbox/source_gen/HamsterSimulatorLanguage/sandbox/HamsterLibrary/Territory.cs
/// <filename>
///     Territory.cs
/// </filename>
using System.Collections.Generic;
using Alf.Library.PrimitiveBehaviors;
using HamsterLibrary;

namespace HamsterLibrary
{
    internal class Territory
    {

        public Territory(int rows, int columns)
        {
            for (int rowIndex = 1; rowIndex <= rows; rowIndex++)
            {
                HamsterLibrary.Row row = new HamsterLibrary.Row(columns);
                this.rows.Add(row);

                if (rowIndex > 1)
                {
                    HamsterLibrary.Row upperRow = this.rows.At(rowIndex - 1 - 1);
                    for (int columnIndex = 1; columnIndex <= columns; columnIndex++)
                    {
                        System.Collections.Generic.List<HamsterLibrary.Cell> helperVar_ylb9x9_a0b0a0d0a0a0_0 = upperRow.cells;
                        HamsterLibrary.Cell upperRowCell = helperVar_ylb9x9_a0b0a0d0a0a0_0.At(columnIndex - 1);
                        HamsterLibrary.Cell currentRowCell = row.cells.At(columnIndex - 1);
                        currentRowCell.north = upperRowCell;
                        upperRowCell.south = currentRowCell;
                    }
                }
            }
        }

        public virtual HamsterLibrary.Cell getCellAt(int x, int y)
        {
            HamsterLibrary.Row row = this.rows.At(y + 1 - 1);
            System.Collections.Generic.List<HamsterLibrary.Cell> helperVar_ylb9x9_b0b0_0 = row.cells;
            HamsterLibrary.Cell cell = helperVar_ylb9x9_b0b0_0.At(x + 1 - 1);
            return cell;
        }

        public System.Collections.Generic.List<HamsterLibrary.Row> rows = new System.Collections.Generic.List<HamsterLibrary.Row>();
    }
}

File: languages/HamsterSimulatorLanguage.sandbox/source_gen/HamsterSimulatorLanguage/sandbox/HamsterLibrary/Territory.cpp
#include "Territory.h"
#include <memory>
#include "Territory.h"
#include <vector>
#include <alf/library/collectionfunctions/CollectionFunctions.hpp>
#include <optional>
#include "Row.h"
#include "Cell.h"

namespace HamsterLibrary
{
  Territory::Territory(int rows, int columns)
  {
    for (int rowIndex = 1; rowIndex <= rows; rowIndex++)
    {
      std::shared_ptr<Row> row = std::make_shared<Row>(columns);
      this->rows.push_back(row);

      if (rowIndex > 1)
      {
        std::optional<std::shared_ptr<Row>> upperRow = alf::library::primitivebehaviors::CollectionFunctions::at(this->rows, rowIndex - 1 - 1);
        for (int columnIndex = 1; columnIndex <= columns; columnIndex++)
        {
          std::vector<std::shared_ptr<Cell>> helperVar_ylb9x9_a0b0a0d0a0a0 = upperRow.has_value() ? upperRow.value()->cells : std::vector<std::shared_ptr<Cell>>();
          std::optional<std::shared_ptr<Cell>> upperRowCell = alf::library::primitivebehaviors::CollectionFunctions::at(helperVar_ylb9x9_a0b0a0d0a0a0, columnIndex - 1);
          std::optional<std::shared_ptr<Cell>> currentRowCell = alf::library::primitivebehaviors::CollectionFunctions::at(row->cells, columnIndex - 1);
          currentRowCell.value()->north = upperRowCell;
          upperRowCell.value()->south = currentRowCell;
        }
      }
    }
  }
  std::optional<std::shared_ptr<Cell>> Territory::getCellAt(int x, int y)
  {
    std::optional<std::shared_ptr<Row>> row = alf::library::primitivebehaviors::CollectionFunctions::at(this->rows, y + 1 - 1);
    std::vector<std::shared_ptr<Cell>> helperVar_ylb9x9_b0b0 = row.has_value() ? row.value()->cells : std::vector<std::shared_ptr<Cell>>();
    std::optional<std::shared_ptr<Cell>> cell = alf::library::primitivebehaviors::CollectionFunctions::at(helperVar_ylb9x9_b0b0, x + 1 - 1);
    return cell;
  }
}
